src/compiler/scala/tools/nsc/transform/Constructors.scala

```diff
@@ -617,8 +620,19 @@ abstract class Constructors extends Statics with Transform with TypingTransforme
           stat match {
             case ValDef(mods, name, _, _) if mods.hasFlag(PRESUPER) => // TODO trait presupers
               // stat is the constructor-local definition of the field value
-              val fields = presupers filter (_.getterName == name)
-              assert(fields.length == 1, s"expected exactly one field by name $name in $presupers of $clazz's early initializers")
+              val fields = presupers.filter { v =>
+                val nm =
+                  if (v.symbol.isPrivateLocal) {
+                    val g = v.getterName
+                    val i = g.lastPos(scala.reflect.NameTransformer.NAME_JOIN_STRING(0))
```

Comment:
This is to undo `expandName` right? Checking `v.symbol.hasFlag(EXPANDEDNAME)` above and splitting the name at `EXPAND_SEPARATOR_STRING` would be a bit more precise?

Reply:
thanks